app/extensions/api/namespace.py

```diff
@@ -199,6 +202,15 @@ def wrapper(*args, **kwargs):
 
         return decorator
 
+    def register_decorator_in_function(self, func, decorator_to_register):
```

Comment:
1. Let's start the name with underscore since it is for internal use only.
2. I think since we only register access-restriction-related decorators, we may indicate this in the function name, e.g. `_register_access_restriction_decorator`.
3. Also, rename `_decorators` to `_access_restriction_decorators`.